src/lib/duration/create.js

```diff
@@ -18,6 +18,7 @@ var isoRegex = /^(-|\+)?P(?:([-+]?[0-9,.]*)Y)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9
 
 export function createDuration (input, key) {
     var duration = input,
+        numericInput = typeof input === 'string' && !isNaN(input) ? +input : input,
```

Comment:
I grabbed this typecasting logic from what powers stringified number arguments in moment().add(...) in an effort to keep the change minimal and not to undo any possible historical reasoning. 

The bizarre behavior of unary `+` makes this a difficult problem. We don't want to run `+input` on non-strings because many other types would result in `isNumber(+input) === true`:
```
+true  // 1
+false // 0
+null  // 0
+[]    // 0
```

We also can't move the `isNaN(input)` check to the if condition due to cases where `input === NaN` relying on that condition for marking dates as invalid. (Changing the if conditions breaks 85 tests concerning invalid durations)